synapse/appservice/api.py

```diff
@@ -136,10 +137,15 @@ async def query_user(self, service: "ApplicationService", user_id: str) -> bool:
             args = None
             if self.config.use_appservice_legacy_authorization:
                 args = {"access_token": service.hs_token}
+
+            headers: Dict[bytes, List[bytes]] = {
```

Comment:
I don't think we need to manually type each of these:

```suggestion
            headers = {
```

(Please apply this everywhere that is applicable.)

I also think it might be clearer to have a single method and just call `headers=self._get_headers()` in each spot?